accepted/exception-handling.md

```diff
@@ -0,0 +1,540 @@
+# Summary
+
+Implement support for the WebAssembly [exception handling
+proposal](https://github.com/WebAssembly/exception-handling).
+
+# Motivation
+[motivation]: #motivation
+
+The WebAssembly exception handling proposal introduces mechanisms for
+raising, catching, and propagating exceptions. The proposal has been
+subject to some controversy as the proposal has undergone several
+significant revisions, with an early revision being deployed in
+production already. As a result there effectively exists two revisions
+of exception handling in the wild at the moment:
+
+* The official W3C WebAssembly exception handling revised proposal;
+* and, the legacy exception handling revision.
+
+The revised proposal was advanced to phase 4 during [a community group
+meeting in
+July](https://github.com/WebAssembly/meetings/blob/main/main/2024/CG-07-16.md). Support
+for exception handling is (at least) of interest to C++, Kotlin, and
+OCaml toolchains. The proposal is also a prerequisite for the [stack
+switching
+proposal](https://github.com/WebAssembly/stack-switching/blob/main/proposals/stack-switching/Explainer.md)
+which plans to use exceptions to finalise stacks. Nonetheless, to
+remain standard compliant, we will eventually have to implement the
+exception handling proposal in Wasmtime.
+
+# Requirements
+[requirements]: #requirements
+
+* **Exception handling should be a zero-cost feature**, i.e. the
+  implementation must not affect the run-time performance (execution
+  speed, memory usage, etc) of programs that do not use the feature.
+
+* **No dependence on `libunwind`.** This library provides a C API for
+  determining and unwinding call chains in ELF programs. Nonetheless,
+  we do not want to bring it into our trusted computing base (TCB),
+  because we have found its implementations to be buggy in the past,
+  and such bugs can easily compromise the integrity of Wasmtime (some
+  past `libunwind` bugs:
+  [#2808](https://github.com/bytecodealliance/wasmtime/issues/2808),
+  [#3256](https://github.com/bytecodealliance/wasmtime/issues/3256),
+  [#7997](https://github.com/bytecodealliance/wasmtime/issues/7997)).
+
+* **Fast unwinding strategy.** We require unwinding to be fast enough
+  to support [the guest
+  profiler](https://docs.wasmtime.dev/examples-profiling-guest.html). Thus,
+  it is important that the chosen implementation strategy does not box
+  ourselves into a corner. Though, we consider it acceptable if the
+  minimal viable product (MVP) of the implementation does not achieve
+  peak performance for unwinding as long as there is a viable path to
+  improving the performance.
```

Comment:
Guest profiling doesn't need stack unwinding, only stack walking, right? The difference between the two is that stack unwinding will run compiled code for each stack frame (and generally destroy the stack frame in the process) while stack walking just iterates over all stack frames and gets the instruction pointer. Stack walking can be done with either frame pointers or unwind tables, while stack unwinding needs more compiler cooperation. Cranelift already supports stack walking.

Reply:
Good clarification :+1: